book/release-service.md

```diff
@@ -39,8 +28,95 @@ The diagram below shows the flow of custom resources and the orchestration of pi
+|-------------------------|--------------------------------------------------------|--------------------------------------------------------------------------|
+| Application & Component | When validating ReleasePlans and ReleasePlanAdmissions | To ensure data consistency                                               |
+| ReleasePlan             | When starting to process a Release                     | To determine the target workspace to run the Release Pipeline            |
+| ReleasePlanAdmission    | When starting to process a Release                     | To determine the Release Strategy to use to compose the Release Pipeline |
+| Snapshot                | Creating and triggering a Release Pipeline             | To determine which components to release                                 |
+| ReleaseStrategy         | Before composing the Release Pipeline                  | To determine the bundle to use to run the Release Pipeline               |
+
+### UPDATE
+
+| Custom Resource            | When?                                                    | Why?                                                         |
+|----------------------------|----------------------------------------------------------|--------------------------------------------------------------|
+| SnapshotEnvironmentBinding | When deploying the Snapshot to an Environment            | Update exiting SnapshotEnvironmentBinding for a new Snapshot |
+| Release                    | During the lifecycle of an attempt to release a Snapshot | Provide status for the execution of the Release Pipeline     |
+
+### WATCH
+
+| Custom Resource              | When?                                                   | Why?                                                            |
+|------------------------------|---------------------------------------------------------|-----------------------------------------------------------------|
+| PipelineRun                  | Once Pipeline created                                   | Relay the Release PipelineRun status to the Release for viewing |
+| SnapshotEnvironmentBinding   | After the SnapshotEnvironmentBinding created or updated | Relay the GitOps deployment status to the Release for viewing   |
+
+#### Annotations/Labels
+
+Following the [annotation guidelines](https://docs.google.com/document/d/1gyXM3pkKFFfxHZnopBi_53vREFWhwA0pFUuIhopDuEo/edit#), the Release Service sets the below annotations on PipelineRuns.
+```
+“pipelines.appstudio.openshift.io/type":     "release",
+"appstudio.openshift.io/snapshot":           “<snapshot name>”,
+“appstudio.openshift.io/component":          “<component name>”,
+“appstudio.openshift.io/application":        “<application name>”,
+"release.appstudio.openshift.io/name":       "<release name>",
+"release.appstudio.openshift.io/namespace":  "<managed workspace name>"
```

Comment:
we use the release namespace for this value. the release is always in the dev workspace, not managed